src/compiler/diagnostic.cc

```diff
@@ -37,6 +37,51 @@ static FILE* text_green(FILE* f) { return f; }
 namespace toit {
 namespace compiler {
 
+const char* const NO_WARN_MARKER = "// NO-WARN";
+
+void Diagnostics::report(Severity severity, const char* format, va_list& arguments) {
+  severity = adjust_severity(severity);
+  bool was_emitted = emit(severity, format, arguments);
+  if (!was_emitted) return;
+  if (severity == Severity::error) encountered_error_ = true;
+  if (severity == Severity::warning) encountered_warning_ = true;
+}
+
+// A hackish way of finding '// NO-WARN' comments.
+// This approach is simple, but doesn't work all the time. Specifically, we might not
+// report warnings in multi-line strings or toitdocs.
+// Example:
+// ```
+// str := """
+//    $(some-warning-operation)  // NO-WARN
+// """
+// In this case the '// NO-WARN' is part of the string and shouldn't be recognized.
+bool Diagnostics::ends_with_no_warn_marker(const Source::Position& pos) {
+  // See if the line ends with the NO_WARN_MARKER.
+  auto manager = source_manager();
+  if (manager == null) return false;
+  auto source = manager->source_for_position(pos);
+  auto text = source->text_at(pos);
+  // Find the end of the line.
+  size_t i = 0;
+  while (text[i] != '\n' && text[i] != '\0')
```

Comment:
Either put this on one line or add braces.

Reply:
done.